src/AddressRegistry/StreetName/ReaddressService.cs

```diff
@@ -0,0 +1,185 @@
+namespace AddressRegistry.StreetName
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Address;
+    using Commands;
+    using DataStructures;
+    using Exceptions;
+
+    public sealed class ReaddressService
+    {
+        private readonly IPersistentLocalIdGenerator _persistentLocalIdGenerator;
+        private readonly IEnumerable<ReaddressAddressItem> _addressesToReaddress;
+        private readonly StreetName _streetName;
+
+        private readonly List<ReaddressedAddressData> _readdressedAddresses;
+        private readonly List<(ReaddressAction, AddressPersistentLocalId)> _actions;
+
+        public ReaddressService(
+            IPersistentLocalIdGenerator persistentLocalIdGenerator,
+            IEnumerable<ReaddressAddressItem> addressesToReaddress,
+            StreetName streetName)
+        {
+            _persistentLocalIdGenerator = persistentLocalIdGenerator;
+            _addressesToReaddress = addressesToReaddress;
+            _streetName = streetName;
+
+            _readdressedAddresses = new List<ReaddressedAddressData>();
+            _actions = new List<(ReaddressAction, AddressPersistentLocalId)>();
+        }
+
+        public ReaddressResult Execute()
```

Comment:
Shouldn't we pass `addressesToReadress` here? That way the service can be more atomic.

Reply:
Also `Execute` is not a good name.
You could do `ReadressFor(IEnumerable<ReaddressAddressItem> addressesToReaddress)` or something like that.